app/scripts/controllers/balance.js

```diff
@@ -0,0 +1,60 @@
+const ObservableStore = require('obs-store')
+const PendingBalanceCalculator = require('../lib/pending-balance-calculator')
+const BN = require('ethereumjs-util').BN
+
+class BalanceController {
+
+  constructor (opts = {}) {
+    const { address, ethStore, txController } = opts
+    this.address = address
+    this.ethStore = ethStore
+    this.txController = txController
+
+    const initState = {
+      ethBalance: undefined,
+    }
+    this.store = new ObservableStore(initState)
+
+    this.balanceCalc = new PendingBalanceCalculator({
+      getBalance: () => Promise.resolve(this._getBalance()),
+      getPendingTransactions: this._getPendingTransactions.bind(this),
+    })
+
+    this.registerUpdates()
+  }
+
+  async updateBalance () {
+    const balance = await this.balanceCalc.getBalance()
+    this.store.updateState({
+      ethBalance: balance,
+    })
+  }
+
+  registerUpdates () {
+    const update = this.updateBalance.bind(this)
+    this.txController.on('submitted', update)
+    this.txController.on('confirmed', update)
+    this.txController.on('failed', update)
+    this.txController.blockTracker.on('block', update)
+  }
+
+  _getBalance () {
+    const store = this.ethStore.getState()
```

Comment:
i like this style recently
```js
 const { accounts } = this.ethStore.getState()
```

Reply:
I'm slightly ambivalent about this, because it destructures and calls a method on the same line, so if there was an error on this line, it would be ambiguous which operation caused it.